src/lib/cloud-provider.js

```diff
@@ -0,0 +1,124 @@
+import log from './log.js';
+
+
+class CloudProvider {
+    constructor (cloudHost, vm, user, projectId) {
```

Comment:
Docs would be good here. Also I would suggest `username` rather than `user` since it could be confusing going between www — in www `user` might be an object.